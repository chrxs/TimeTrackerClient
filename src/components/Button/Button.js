import React from 'react'
import PropTypes from 'prop-types'

import styles from './Button.scss'

const Button = ({
  type,
  label,
  disabled,
  onClick,
  ...otherProps
}) => {
  return (
    <button
      type={type}
      className={styles.Button}
      disabled={disabled}
      onClick={onClick}
      {...otherProps}
    >
      { label }
    </button>
  )
}

Button.displayName = 'Button'

Button.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func
}

Button.defaultProps = {
  type: 'button',
  disabled: false,
  onClick: () => {}
}

export default Button
