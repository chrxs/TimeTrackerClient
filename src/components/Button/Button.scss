.Button {
  background: #df4a32;
  border: none;
  border-radius: 3px;
  color: #fff;
  cursor: pointer;
}
