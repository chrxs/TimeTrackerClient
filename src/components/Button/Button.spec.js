/* global describe, test, expect */
import React from 'react'
import Button from './Button'
import renderer from 'react-test-renderer'

describe('Button snapshots', () => {
  test('with just label prop', () => {
    const component = renderer.create(<Button label='Do Something' />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with all props', () => {
    const component = renderer.create(<Button label='Do everything' type='button' disabled={false} onClick={() => {}} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
