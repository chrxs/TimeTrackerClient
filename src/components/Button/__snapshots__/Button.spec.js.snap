// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshots with all props 1`] = `
<button
  className="Button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Do everything
</button>
`;

exports[`Button snapshots with just label prop 1`] = `
<button
  className="Button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Do Something
</button>
`;
